remix-lib/src/helpers/txResultHelper.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+const { bufferToHex, isHexString } = require('ethereumjs-util')
+
+function convertToPrefixedHex (input) {
+  if (input === undefined || input === null || isHexString(input)) {
+    return input
+  } else if (Buffer.isBuffer(input)) {
+    return bufferToHex(input)
+  }
+  return '0x' + input.toString(16)
+}
+
+/*
+ txResult.result can be 3 different things:
+ - VM call or tx: ethereumjs-vm result object
+ - Node transaction: object returned from eth.getTransactionReceipt()
+ - Node call: return value from function call (not an object)
+
+ Also, VM results use BN and Buffers, Node results use hex strings/ints,
+ So we need to normalize the values to prefixed hex strings
+*/
+function resultToRemixTx (txResult) {
+  const { result, transactionHash } = txResult
+  const { status, vm, gasUsed, createdAddress, contractAddress } = result
+  let returnValue, errorMessage
+
+  if (isHexString(result)) {
```

Comment:
I don't think there will be any case satisfying this case. I will suggest to cover all branches with the tests, that can help to track these kind of issues.

Reply:
Do you mean the if(isHexString(result) {} part isn't covered? It's covered by the 'node call result' test.

Side note: `result` is always hex string when you call a view method (except in the VM). So, in that case, all the destructured variables like status, vm, etc. are all undefined because result is a string and not an object. It's not great, but it works. I'm open to better ways of doing this though